### PR TITLE
Feature/check user reference

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### This PR adds | fixes:

- Refactor check_user_reference

### How to test:

- Automatic and continuous test suite

### Expected outcome:
- [ ] Installation, unit and integration test suite pass

### Review:
- [ ] Code review
- [ ] Tests pass

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes. If applicable record manual test results in PR header
- [ ] **MINOR** - when you add functionality in a backwards compatible manner. If applicable record manual test results in PR header
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: lib/MIP/Download.pm [hunk 1]
```diff
@@ -91,20 +88,19 @@ sub check_user_reference {
...
 
-                ## Found match
-                if (
-                    exists $reference_ref->{$reference_id}{$reference_genome_version}
-                    {$version} )
-                {
+                my $has_attribute = get_download_reference_attributes(
+                    {
+                        id             => $reference_id,
+                        genome_version => $reference_genome_version,
```
Comment (jemten) [inline lib/MIP/Download.pm:94]:
```suggestion
                        genome_version => $reference_genome_version,
                        id             => $reference_id,
```

File: lib/MIP/Download.pm [hunk 2]
```diff
@@ -113,14 +109,66 @@ sub check_user_reference {
...
+## Arguments: $id             => Reference id
+##          : $genome_version => Reference genome build versions
+##          : $reference_href => Reference id and reference version per genome version
+##          : $version        => Reference version
+
+    my ($arg_href) = @_;
+
+    ## Flatten argument(s)
+    my $id;
+    my $genome_version;
```
Comment (jemten) [inline lib/MIP/Download.pm:134]:
```suggestion
    my $genome_version;
    my $id;
```

Comment (henrikstranneheim) [reply lib/MIP/Download.pm]: Done

File: lib/MIP/Download.pm (hunk 1 above)
Comment (henrikstranneheim) [reply lib/MIP/Download.pm]: Done

**Pull request merged**